janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/GraphCentricQueryBuilder.java

```diff
@@ -242,6 +239,9 @@ public GraphCentricQuery constructQueryWithoutProfile(final ElementCategory resu
         Preconditions.checkNotNull(resultType);
         if (limit == 0) return GraphCentricQuery.emptyQuery(resultType);
 
+        if (this.globalConstraints.isEmpty()) {
```

Comment:
What is the reason for this change?

Reply:
This is because now `JanusGraphStep` calls `constructQueryWithoutProfile` directly, while previously, it calls `constructQuery` which in turn calls `constructQueryWithoutProfile`. Thus I have to move this logic. Anyway, I think the right place for this logic is indeed the `constructQueryWithoutProfile` method.

The reason for calling `constructQueryWithoutProfile` directly is because we don't want the graph centric query to be profiled. `JanusGraphStep` always 1) tries to construct a single graph centric query and see if it can fulfill the constraints. If not, it 2) constructs a few graph centric queries and combines results together. If Step 1) does not succeed, we don't want it to appear in the profiler result.